File: components/sdk/dotnet/src/Spas.Sdk.Metadata/Builders/SecurityBuilder.cs
using Spas.Sdk.Metadata.Models;

namespace Spas.Sdk.Metadata.Builders;

/// <summary>
/// Builder for security metadata.
/// </summary>
public class SecurityBuilder
{
    private string? _authenticationType;
    private readonly List<string> _requiredScopes = new();
    private readonly List<string> _dataClassification = new();

    /// <summary>
    /// Sets the authentication type (e.g., jwt).
    /// </summary>
    public SecurityBuilder WithAuthenticationType(string type)
    {
        _authenticationType = type;
        return this;
    }

    /// <summary>
    /// Adds a required scope.
    /// </summary>
    public SecurityBuilder AddRequiredScope(string scope)
    {
        _requiredScopes.Add(scope);
        return this;
    }

    /// <summary>
    /// Adds a data classification.
    /// </summary>
    public SecurityBuilder AddDataClassification(string classification)
    {
        _dataClassification.Add(classification);
        return this;
    }

    /// <summary>
    /// Builds the security metadata.
    /// </summary>
    public SecurityMetadata Build()
    {
        return new SecurityMetadata
        {
            Authentication = !string.IsNullOrEmpty(_authenticationType) || _requiredScopes.Any()
                ? new AuthenticationMetadata
                {
                    Type = _authenticationType ?? "jwt",
                    RequiredScopes = _requiredScopes
                }
                : null,
            DataClassification = _dataClassification
        };
    }
}

File: components/sdk/java/spas-sdk-metadata/src/main/java/io/spas/sdk/metadata/builders/SecurityBuilder.java
package io.spas.sdk.metadata.builders;

import io.spas.sdk.metadata.model.Authentication;
import io.spas.sdk.metadata.model.AuthType;
import io.spas.sdk.metadata.model.DataClassification;
import io.spas.sdk.metadata.model.Security;

import java.util.ArrayList;
import java.util.List;

/**
 * Fluent builder for security metadata.
 */
public final class SecurityBuilder {
    private AuthType authenticationType;
    private final List<String> requiredScopes = new ArrayList<>();
    private List<DataClassification> dataClassification;
    
    private SecurityBuilder() {}
    
    public static SecurityBuilder create() {
        return new SecurityBuilder();
    }
    
    public SecurityBuilder withAuthenticationType(AuthType type) {
        this.authenticationType = type;
        return this;
    }
    
    public SecurityBuilder addRequiredScope(String scope) {
        this.requiredScopes.add(scope);
        return this;
    }
    
    public SecurityBuilder addDataClassification(DataClassification classification) {
        if (this.dataClassification == null) {
            this.dataClassification = new ArrayList<>();
        }
        this.dataClassification.add(classification);
        return this;
    }
    
    public Security build() {
        Authentication authentication = null;
        if (authenticationType != null || !requiredScopes.isEmpty()) {
            authentication = new Authentication(
                authenticationType,
                requiredScopes.isEmpty() ? null : new ArrayList<>(requiredScopes)
            );
        }
        
        return new Security(
            authentication, 
            dataClassification == null || dataClassification.isEmpty() ? null : dataClassification
        );
    }
}
